osc-server/src/test/java/org/osc/core/broker/util/crypto/X509TrustManagerFactoryTest.java

```diff
@@ -0,0 +1,157 @@
+
+import static org.junit.Assert.*;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.ArrayUtils;
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.osc.core.broker.service.api.server.ArchiveApi;
+import org.osc.core.broker.service.api.server.LoggingApi;
+import org.osc.core.broker.service.archive.ArchiveUtil;
+
+public class X509TrustManagerFactoryTest {
+
+    private static final String TEST_CERT_ALIAS = "testcertificate";
+
+    private static final String RESOURCE_FOLDER = Paths.get(".", "src", "test", "resources").toString() + File.separator;
+
+    private static final String TEST_TRUSTSTORE_FILE_NAME = "osctrustore.jks";
+    private static final String TEST_X509PEM_ZIP_FILE_NAME = "oscx509test_x509pem.zip";
+    private static final String TEST_PKICHAIN_ZIP_FILE_NAME = "oscx509test.zip";
+
+    private static final String TEST_CERT_FILE_NAME = RESOURCE_FOLDER + "testcertificate.crt";
+    private static final String TEST_PRIVATE_KEY_FILE_NAME = RESOURCE_FOLDER + "oscx509test.pem";
+
+    private static final File TEST_CERT_FILE = new File(TEST_CERT_FILE_NAME);
+    private static final File TEST_TRUSTSTORE_FILE = new File(TEST_TRUSTSTORE_FILE_NAME);
+    private static final File TEST_PKICHAIN_ZIP_FILE = new File(TEST_PKICHAIN_ZIP_FILE_NAME);
+    private static final File TEST_X509PEM_ZIP_FILE = new File(TEST_X509PEM_ZIP_FILE_NAME);
+    private static final File TEST_PRIVATE_KEY_FILE = new File(TEST_PRIVATE_KEY_FILE_NAME);
+    private static final char[] JKS_PASSWD = "abc12345".toCharArray();
+
+    @Mock
+    private LoggingApi logging;
+
+    @Spy
+    @InjectMocks
+    private ArchiveApi archiveUtil = new ArchiveUtil();
+
+    @InjectMocks
+    private X509TrustManagerFactory factory;
+
+    @Before
+    public void setup() throws IOException {
+        // These three files would be deleted by the method call.
+        FileUtils.copyFile(new File(RESOURCE_FOLDER + TEST_TRUSTSTORE_FILE_NAME), TEST_TRUSTSTORE_FILE);
+        FileUtils.copyFile(new File(RESOURCE_FOLDER + TEST_PKICHAIN_ZIP_FILE_NAME), TEST_PKICHAIN_ZIP_FILE);
+        FileUtils.copyFile(new File(RESOURCE_FOLDER + TEST_X509PEM_ZIP_FILE_NAME), TEST_X509PEM_ZIP_FILE);
+
+        this.factory = X509TrustManagerFactory.getInstance();
+
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @After
+    public void tearDown() {
+        if (TEST_TRUSTSTORE_FILE.exists()) {
```

Comment:
do we need tear down since the files will be deleted anyway?

Reply:
The zip files will be deleted. The truststore better not be.